### PR TITLE
fix[WindowsHierarchy]: task #865buw83b

### PR DESCRIPTION
**Pull request by Dedado3000**

### CODE REVIEW

**Comment by PaablooCH** [comment]
Task linked: [CU-865buw83b Bug: Destruction of a Father of selected GameObject](https://app.clickup.com/t/865buw83b)

**Comment by markmeliaa** [comment]
Another better solution found.

File: Source/DataModels/Windows/EditorWindows/WindowHierarchy.cpp [hunk 1]
```diff
@@ -112,7 +112,8 @@ void WindowHierarchy::DrawRecursiveHierarchy(GameObject* gameObject)
         {
             if (ImGui::MenuItem("Delete"))
             {
-                if (gameObject == App->scene->GetSelectedGameObject())
```
Comment (gerard-queralt) [inline Source/DataModels/Windows/EditorWindows/WindowHierarchy.cpp:115]: Wouldn't it be more efficient to also leave this check, even though it's redundant? This way we would avoid traversing the vector when we are deleting the selected game object, I think

**Comment by gerard-queralt** [comment]
Nevermind then :P

**Comment by markmeliaa** [comment]
> Nevermind then :P

Sorry! We are working on it right now.

**Pull request closed without merging**